portfolio/src/main/webapp/script.js

```diff
@@ -106,23 +90,27 @@ function start() {
         }, 1000);
 
         currentQuestion += 1;
-        if (currentQuestion >= questions.length) {
-            let endText = document.createElement("div");
-            endText.innerText = `Your score: ` + score +
-                `/` + questions.length + `. Thank you for playing! By the way, that's true, the right answer is always right :)`;
-            endText.classList.add("quizEnd");
-            quizContainer.innerHTML = "";
-            quizContainer.append(endText);
-            return;
-        }
-        scoreElement.innerHTML = "Current score: " + score;
-        question.innerHTML = questions[currentQuestion];
-        leftButton.innerHTML = leftAnswers[currentQuestion];
-        rightButton.innerHTML = rightAnswers[currentQuestion];
+        nextQuestion(currentQuestion, questions, leftAnswers, rightAnswers, score, scoreElement, leftButton, rightButton, question, quizContainer);
     });
+    nextQuestion(currentQuestion, questions, leftAnswers, rightAnswers, score, scoreElement, leftButton, rightButton, question, quizContainer);
+}
 
+function nextQuestion(currentQuestion, questions, leftAnswers, rightAnswers, score, scoreElement, leftButton, rightButton, question, quizContainer) {
+    if (currentQuestion >= questions.length) {
+        endGame(score, questions, quizContainer);
+        return;
+    }
     scoreElement.innerHTML = "Current score: " + score;
     question.innerHTML = questions[currentQuestion];
     leftButton.innerHTML = leftAnswers[currentQuestion];
     rightButton.innerHTML = rightAnswers[currentQuestion];
 }
+
+function endGame(score, questions, quizContainer) {
+    let endText = document.createElement("div");
```

Comment:
const endText may be better, given that you never reasign it

Reply:
Okay, thank you! Done.